src/lib/utils.ts

```diff
@@ -288,17 +289,52 @@ export function encodeKey(
             `Incorrect value for tuple. Got: ${value}, expected array.`,
           );
         }
+
+        const isCompactBytesArray: boolean = schema.valueType.includes(
+          '[CompactBytesArray]',
+        );
+
+        if (Array.isArray(value[0]) && isCompactBytesArray) {
+          const valueType = schema.valueType.replace('[CompactBytesArray]', '');
+          const valueContent = schema.valueContent.replace(
+            '[CompactBytesArray]',
+            '',
+          );
+
+          const encodedTuples = value.map((element) => {
+            return encodeTupleKeyValue(valueContent, valueType, element);
+          });
+          return encodeValueType('bytes[CompactBytesArray]', encodedTuples);
+        }
+
         return encodeTupleKeyValue(
           schema.valueContent,
           schema.valueType,
           value,
         );
       }
 
+      // This adds an extra check to ensure the casting below is safe
+      // TODO: refactor to fix the TS typing.
+      if (
+        Array.isArray(value) &&
+        Array.isArray(
+          value[0] && !isValidTuple(schema.valueType, schema.valueContent),
+        )
```

Comment:
why do you put '&& !isValidTuple(schema.valueType, schema.valueContent)' inside the `isArray` ??

Reply:
I think you want this:

```suggestion
        Array.isArray(value[0]) &&
        !isValidTuple(schema.valueType, schema.valueContent),
```